src/haxelib/Data.hx

```diff
@@ -190,6 +190,8 @@ typedef DefineDocumentation = {
 	var Bsd = 'BSD';
 	var Public = 'Public';
 	var Apache = 'Apache';
+	var Mpl = 'MPL';
+	var Other = "Other";
```

Comment:
The reason there is no `Other` license currently is because that could include non open-source licenses. Libraries uploaded to `lib.haxe.org` should be open source. For libraries with another license, it is possible to set the license field to `Unknown` and this works fine for most use cases, just not for publishing to lib.haxe.org.

Reply:
In that case would something like `OtherOpenSource` work? The name might be a bit long but I do think there should be an option to specify other (custom) licenses.

Yes, the possibility to have projects with non open-source licenses be published is possible, but I don't think we should restrict the developer because of what *should* be uploaded to haxelib.

Reply:
Sure, I understand, but currently there are library authors opting for a license that doesn't align with their actual project. They're forced to misrepresent their library on lib.haxe.org and I think that situation is worse than the very unlikely scenario of someone submitting non open-source libraries. 

Unknown is not appropriate here either.

There is an evolving open source licensing landscape as well as custom licensing priorities that should be considered.

Reply:
@tobil4sk 

I updated the change to create better clarification. I think we should address this as it is a real problem. Misrepresented haxelib licenses create an unfavorable scenario here. Perhaps this isn't the perfect solution, but I'm happy to start the conversation.

The made these changes when I discovered that the license listed for this library was in fact not Public Domain as it is listen on haxelib, so I hope you can understand the frustration of both library consumers and publishers.

![6e9344edd368731dd40ad0baee13fb67](https://github.com/user-attachments/assets/a479035c-d73d-416b-82de-d4bcbf33311b)

https://github.com/polybiusproxy/hxCodec

Reply:
> I originally made these changes when I discovered that the license listed for this library(Link below) was in fact not Public Domain as it is incorrectly listed as on haxelib, so I hope you can understand the frustration of both library consumers and publishers.

Yes I agree, this is a bad scenario and something should be done to prevent this in future. However, as I understand it, the reason to not add a custom license field is precisely to avoid openly welcoming more problems like this where licenses are unclear and users might be unaware of a (possibly proprietary or unspecified) license of code they are using. Though not ideal, the current situation has at least compelled the library author to explicitly bring attention to the fact that the license is different. This is a rare case though, and in most cases the strict requirement has encouraged use of one of the existing choices (over uploading unlicensed code for example).

I think it would be better to add MPL and possibly other licenses individually if there is a good reason why an author can't use any of the existing ones. Maybe we can add something to the error to encourage authors to get in touch if they need a license that's not available, to help prevent these issues in future.

If we were to add a custom license option, open source is a pretty vague definition, and if anything it would be better to go for something more concretely defined like "OSI-approved".

Here are some links to previous discussions on this topic:
 - https://github.com/HaxeFoundation/haxelib/issues/178#issuecomment-78847937
 - https://github.com/HaxeFoundation/haxelib/issues/52
 
@Simn What are your thoughts?

Reply:
Great idea! I think a `OSI-Improved` would be a very welcome additon.

Reply:
I don't see OSI-approved being a valid approach because people may use this option to misrepresent their license too. For example, I use a software license for a particular work that is based on MPL 2.0 that follows MPL's rules for modifying the license to add a clause or addendum. It has not been OSI approved (even it is still an osi compliant license). Furthermore, all of the other licenses currently supported are OSI licenses anyway, so it seems a bit redundant and does not necessarily resolve the issue, imo.

Reply:
I see. Perhaps OSI-Other would be more appropriate. All of the currently accepted licenses are OSI approved **already** so it might be a little redundant. More importantly... I think any label that encourages actually checking the license included in the library would be beneficial here. We don't want a system that encourages dishonesty, simply because the mechanisms don't allow for legitimate use.

Personally, I don't see the point of a restriction in the first place. Google Code had a similar restriction and it was heavily referenced in these decade-old discussions. Google Code is dead and competitors like Github have taken over the landscape where there are no such limitations. It all seems a bit silly to me.

Reply:
Ok! I suppose there really aren't any new standard licenses coming out of the woodwork (that I am aware of at least). Maybe it's ok if we just add MPL for now after all. I don't actually see the point of restricting non-standard, "shared-source" or academic licenses, but I think at least adding MPL closes the gap.

Made some final changes to my PR to narrow it down to just Mozilla Public License. Thanks for the feedback!